runtime/compiler/optimizer/J9ValuePropagation.cpp

```diff
@@ -278,6 +278,7 @@ bool J9::ValuePropagation::transformIndexOfKnownString(
    // Keep track of whether or not all constraints are global.
    bool isGlobal = true;
    bool isGlobalQuery;
+   int32_t headerSize = TR::Compiler->om.isOffHeapAllocationEnabled() ? 0 : TR::Compiler->om.contiguousArrayHeaderSizeInBytes();
```

Comment:
appeared wrong and regressed to me:
1) hardcoding the so-called headersize is not a good approach;
2) headerSize is different from arrayBaseOffset to begin with.  In off-heap, headerSize is not zero;
3) adding 0 here and subtracting real off-heap header-size in the callee is exactly wrong. The original code was right to me.

Reply:
Forgot to remove the subtraction of headerSize, fixed.
Updated the naming and fixed comment in `getAddressOfElement` 

Reason being, typically the helper `getAddressOfElement` assumes headerSize is included in the offset from the array access trees or explicitly adds it for other cases (folding String.indexOf). With offheap generated trees for array access do not have headerSize addition hence breaks the helper assumption in those cases without explicit addition.

Modified code assumes no headerSize is included in the offset when offheap is used, updating the `String.indexOf` case, and consistent with array access trees and correct if Unsafe access uses the helper (Unsafe.arrayBaseOffset value being 0 for offheap).